### PR TITLE
Made the changes to readme and tests to accurately depict go rules

### PR DESCRIPTION
**Pull request by jhuiet**
I made this change to address the way the go-counting application was structured. The tests weren't consistent with the rules. Small change, addressing an issue I posted earlier (check issue go-counting for more detail)

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/go-counting/README.md [hunk 1]
```diff
@@ -22,8 +23,8 @@ marked "I" (ignored).  Empty spaces represent empty intersections.
 +----+
 |IOOI|
 |O  O|
-|O OI|
-|IOI |
+|O O |
```
Comment (sjwarner-bp) [inline exercises/go-counting/README.md]: The README is generated using `configlet` based on [this description](https://github.com/exercism/problem-specifications/blob/master/exercises/go-counting/description.md)

From what I can see, this README did not need to change :slightly_smiling_face:

File: exercises/go-counting/src/test/java/GoCountingTest.java [hunk 2]
```diff
@@ -12,167 +12,167 @@
...
 	public void whiteCenter5x5BoardTest() {
 		GoCounting gocounting = new GoCounting(board5x5);
-		
+
 		Set<Point> territory = new HashSet<>();
 		territory.add(new Point(2, 3));
-		
-		assertEquals(Player.WHITE, gocounting.getTerritoryOwner(2, 3));
+
+		assertEquals(Player.NONE, gocounting.getTerritoryOwner(2, 3));
```
Comment (sjwarner-bp) [inline exercises/go-counting/src/test/java/GoCountingTest.java]: It looks like this Player should be `WHITE` according to the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/go-counting/canonical-data.json) so I don't think this test needs to change at the minute :slightly_smiling_face:

File: exercises/go-counting/src/test/java/GoCountingTest.java [hunk 3]
```diff
@@ -12,167 +12,167 @@
...
-		
+
 		Set<Point> territory = new HashSet<>();
 		territory.add(new Point(0, 3));
 		territory.add(new Point(0, 4));
 		territory.add(new Point(1, 4));
-		
-		assertEquals(Player.NONE, gocounting.getTerritoryOwner(1, 4));
+
+		assertEquals(Player.BLACK, gocounting.getTerritoryOwner(1, 4));
```
Comment (sjwarner-bp) [inline exercises/go-counting/src/test/java/GoCountingTest.java]: It looks like this Player should be `NONE` according to the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/go-counting/canonical-data.json) so I don't think this test needs to change at the minute :slightly_smiling_face:

**Comment by charles-wangkai** [comment]
@jhuiet Seems that you're seeing Point(x, y) as the point of x-th row and y-th column. But if it's reversed as the point of x-th column and y-th row, that is, x-axis is horizontal and y-axis is vertical, the original test cases were all correct.

**Comment by FridaTveit** [comment]
Thanks for wanting to help improve our exercises @jhuiet! 🙂 

As discussed in the issue you opened (#1395), this change should be made in the [problem specifications repo](https://github.com/exercism/problem-specifications). So I'll close this PR for now, and if you think there is something we could improve in this exercise then please open an issue or a PR in problem specifications 🙂

**Pull request closed without merging**